File: common/src/main/java/xyz/chunkstories/util/VersionInfo.java
//
// This file is a part of the Chunk Stories Implementation codebase
// Check out README.md for more information
// Website: http://chunkstories.xyz
//

package xyz.chunkstories.util;

public class VersionInfo {
	public static String version = autoVersion();
	public static int networkProtocolVersion = 37;

	private static String autoVersion() {

		// If compiled jar, it has this
		String ver = VersionInfo.class.getPackage().getImplementationVersion();
		if (ver != null)
			return ver;

		return "Debug build";
	}
}

File: common/src/main/java/xyz/chunkstories/util/VersionInfo.kt
//
// This file is a part of the Chunk Stories Implementation codebase
// Check out README.md for more information
// Website: http://chunkstories.xyz
//

package xyz.chunkstories.util

import com.google.gson.Gson
import java.io.File

object VersionInfo {
    val versionJson: VersionJson
    val networkProtocolVersion = 37

    init {
        val file = File("version.json")

        versionJson = if (file.exists()) {
            val gson = Gson()
            gson.fromJson(file.readText(), VersionJson::class.java)!!
        } else {
            VersionJson("unknown", "unknown(no version.json)", "unknown", "unknown")
        }
    }
}

data class VersionJson(val version: String, val verboseVersion: String, val commit: String, val buildtime: String)